### PR TITLE
Task Revival endpoint

### PR DESCRIPTION
**Pull request by youliangtan**
## Feature
This will add a `/revive_task` endpoint to the current `rmf_demos_panel` `api-server`. This will enable user to revive a previously canceled/failed task.

## How to test
 - `rmf_ros2`: developing branch:  `feature/task-revival`

Compile, then follow the steps below:
1) Run rmf_demos office world as normal
```bash
ros2 launch rmf_demos office.launch.xml
```
2) Open localhost:5000, then submit task list  `task_lists/office_tasks.json`
3) Cancel Task: cancel `Loop2` on the Web panel
4) Revive the Canceled Task:
```bash
curl -X POST http://localhost:8080/revive_task -d '{"task_id": "Loop2"}' -H "Content-Type: application/json"
```

Signed-off-by: youliang <tan_you_liang@hotmail.com>

### CODE REVIEW

**Comment by vallq** [comment]
hi youliang, I was unable to revive the task based on the instructions above, kept receiving the `[api_server]: /revive_task srv call failed` message. tried calling the endpoint with the `{"task_id": "Loop2"}` on Postman as well but the received boolean is still `false`.
I rebuilt all the packages in the workspace too just to make sure I didn't leave anything out

File: rmf_demos_panel/rmf_demos_panel/api_server.py [hunk 1]
```diff
@@ -74,6 +74,15 @@ def cancel():
     return jsonify({"success": cancel_success})
 
 
+@app.route('/revive_task', methods=['POST'])
+def revive():
+    id = request.json['task_id']
+    is_success = dispatcher_client.revive_task_request(id)
+    logging.debug(f" ROS Time: {dispatcher_client.ros_time()} | \
+        Cancel Task: {id}, success: {is_success}")
```
Comment (vallq) [inline rmf_demos_panel/rmf_demos_panel/api_server.py:82]: minor: also I noticed that this is still `Cancel` and not `Revive`

File: rmf_demos_panel/rmf_demos_panel/api_server.py [hunk 2]
```diff
@@ -74,6 +74,15 @@ def cancel():
     return jsonify({"success": cancel_success})
 
 
+@app.route('/revive_task', methods=['POST'])
+def revive():
+    id = request.json['task_id']
+    is_success = dispatcher_client.revive_task_request(id)
+    logging.debug(f" ROS Time: {dispatcher_client.ros_time()} | \
```
Comment (vallq) [inline rmf_demos_panel/rmf_demos_panel/api_server.py:81]: minor: also I noticed that this is still `Cancel` and not `Revive`

File: rmf_demos_panel/rmf_demos_panel/api_server.py (hunk 1 above)
Comment (vallq) [inline rmf_demos_panel/rmf_demos_panel/api_server.py:82]: minor: also I noticed this is still `Cancel` and not `Revive`

**Review by vallq — COMMENTED** [review]
hi youliang, I was unable to replicate the revival based on the instructions. kept getting the `[api_server]: /revive_task srv call failed` message on the console, and tried to call the API with Postman as well, but the boolean received is still `False`. not sure what I might have done wrong but I rebuilt the entire ws just to be sure

**Comment by tanyouliang95** [comment]
@vallq Are you using the `feature/task-revival` branch from `rmf_ros2`? Also, take note that the task can only be revived if the task was `canceled` or `failed`.

**Comment by luca-della-vedova** [comment]
Closing since `rmf_demos_panel` has long since been deprecated and removed

**Pull request closed without merging**